test/WebSites/ModelBindingWebSite/Controllers/FormCollectionController.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Http;
+using System.IO;
+
+namespace ModelBindingWebSite.Controllers
+{
+    public class FormCollectionController : Controller
+    {
+        public IList<string> ReturnValuesAsList(IFormCollection form)
+        {
+            if (!ModelState.IsValid || form == null)
+            {
+                return null;
+            }
+
+            var valuesList = new List<string>();
+
+            valuesList.Add(form["field1"]);
+            valuesList.Add(form["field2"]);
+
+            return valuesList;
+        }
+
+        public ActionResult ReturnFileContent(IFormCollection form)
+        {
+            if (!ModelState.IsValid || form == null)
+            {
+                return null;
+            }
```

Comment:
if something's obviously not going to plan in a functional test it's better to just blow up (throw) rather than be silent